include/Gaffer/PlugAlgo.h

```diff
@@ -71,6 +71,12 @@ bool canPromote( const Plug *plug, const Plug *parent = NULL );
 /// related to plug layouts are copied to the external
 /// plug - this can be controlled with the `excludeMetadata`
 /// argument.
+///
+/// > Note : When promoting to a Box, this function examines
+/// > the value of the "nodule:type" metadata, and automatically
+/// > creates an intermediate BoxIO node if the plug has a
+/// > nodule. This makes the exact effects of the function dependent
+/// > on whether or not the UI metadata has been loaded.
```

Comment:
Yeah, I do have my concerns over that, but the fact that everything is treated the same and we have a chance to respond to nodule/no-nodule changes on the fly does give it a certain allure. I wonder if someone at that end could do a quick survey to see what worst ratio of non-nodule to nodule plugs is in a production box, and also the ratio of promoted plugs to non-promoted plugs?